node-drainer/pkg/customdrain/client.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package customdrain
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/nvidia/nvsentinel/node-drainer/pkg/config"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+	"sigs.k8s.io/yaml"
+)
+
+type Client struct {
+	dynamicClient dynamic.Interface
+	restMapper    *restmapper.DeferredDiscoveryRESTMapper
+	template      *template.Template
+	config        config.CustomDrainConfig
+}
+
+func NewClient(
+	cfg config.CustomDrainConfig,
+	dynamicClient dynamic.Interface,
+	restMapper *restmapper.DeferredDiscoveryRESTMapper,
+) (*Client, error) {
+	templatePath := filepath.Join(cfg.TemplateMountPath, cfg.TemplateFileName)
+
+	templateContent, err := os.ReadFile(templatePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read template file %s: %w", templatePath, err)
+	}
+
+	tmpl, err := template.New("drain-template").Parse(string(templateContent))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse template: %w", err)
+	}
+
+	gvk := schema.GroupVersionKind{
+		Group:   cfg.ApiGroup,
+		Version: cfg.Version,
+		Kind:    cfg.Kind,
+	}
+
+	_, err = restMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
```

Comment:
Can we fail fast here so that the user knows that something is wrong at startup rather than waiting for drains to fail?